org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -466,10 +448,14 @@ private void handleEvent(DebugEvent debugEvent) {
             // do nothing.
         } else if (event instanceof VMDeathEvent || event instanceof VMDisconnectEvent) {
```

Comment:
VMDeathEvent and VMDisconnectEvent have different meanings and indicate different stages of termination. Please handle them in different code branch.